src/Parallel/Algorithm.hpp

```diff
@@ -3,24 +3,54 @@
 
 #pragma once
 
-#include <functional>
-
-#include "DataStructures/DataBox/DataBox.hpp"
+// <array> wanted for tuple_size
+// IWYU pragma: no_include <array>
+#include <boost/variant/variant.hpp>
+#include <cstddef>
+#include <exception>
+#include <initializer_list>
+#include <lrtslock.h>
+#include <ostream>
+#include <pup.h>
+#include <tuple>
+#include <unordered_set>
+#include <utility>
+
+#include "ErrorHandling/Assert.hpp"
 #include "ErrorHandling/Error.hpp"
-#include "Parallel/ArrayIndex.hpp"
+// Include... ourself?
+// IWYU pragma: no_include "Parallel/Algorithm.hpp"
 #include "Parallel/CharmRegistration.hpp"
 #include "Parallel/ParallelComponentHelpers.hpp"
 #include "Parallel/TypeTraits.hpp"
 #include "Utilities/BoostHelpers.hpp"
-#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ForceInline.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/NoSuchType.hpp"
 #include "Utilities/Overloader.hpp"
 #include "Utilities/PrettyType.hpp"
+#include "Utilities/Requires.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 #include "Utilities/TypeTraits.hpp"
 
+/// \cond
+namespace Parallel {
+template <class Metavariables>
+class CProxy_ConstGlobalCache;
+template <class Metavariables>
+class ConstGlobalCache;
+namespace Algorithms {
+struct Nodegroup;
+struct Singleton;
+}  // namespace Algorithms
+}  // namespace Parallel
+namespace db {
+template <typename TagList>
+class DataBox;
```

Comment:
I guess one thing we'll need to decide is where we want to ultimately include `DataBox`. The final `cpp` file for the executable? The Actions also would need the full definition, so maybe there? Here is also not unreasonable since it's a member of AlgorithmImpl. Just something we should decide on :)

Reply:
On closer inspection removing it from here was incorrect in any case, since AlgorithmImpl adds an explicit `db::DataBox<tmpl::list<>>` to the boost::variant so it should be a complete type.